### PR TITLE
Prepare for Coprocessor Executor Benchmarks

### PR DESCRIPTION
**Pull request by breeswish**
## What have you changed? (mandatory)

This PR sets up foundations for executor benchmarks. As an illustration, it also contains a simple executor benchmark "bench_table_scan_primary_key", which measures time of iterating (by calling `TableScanExecutor::next()`) a row with only primary key listed in interested column. This will happen for SQLs like `SELECT COUNT(*)`. 

Note that this is a layered benchmark, which depends on https://github.com/tikv/tikv/pull/3685

- [x] Wait https://github.com/tikv/tikv/pull/3685 to be merged

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## Benchmark result if necessary (optional)

```
table_scan_primary_key  time:   [21.910 us 22.869 us 25.453 us]
```

### CODE REVIEW

File: benches/coprocessor_executors.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    meta.set_table_id(table.id);
+    meta.set_desc(false);
+    meta.mut_columns().push(id.get_column_info());
+
+    b.iter_with_setup(
+        || {
+            let mut executor = TableScanExecutor::new(
+                meta.clone(),
+                vec![table.get_select_range()],
+                store.clone().into_fixture_store(),
```
Comment (DorianZheng) [inline benches/coprocessor_executors.rs]: Maybe we don't really need a `SimpleBTreeMap` and we can simply implement `into_fixture_store` for `MvccTransactionalStore`?

File: components/test_coprocessor/src/column.rs [hunk 2]
```diff
@@ -13,22 +13,38 @@
...
 
-#[derive(Clone, Copy)]
+#[derive(Clone)]
 pub struct Column {
     pub id: i64,
     pub col_type: i32,
     // negative means not a index key, 0 means primary key, positive means normal index key.
     pub index: i64,
-    pub default_val: Option<i64>, // TODO: change it to Vec<u8> if other type value is needed for test.
+    pub default_val: Option<Datum>, // TODO: change it to Vec<u8> if other type value is needed for test.
```
[the remark this answers is not in the document]
Comment (breeswish) [reply components/test_coprocessor/src/column.rs]: Right!

**Comment by DorianZheng** [comment]
@breeswish Thank you for the PR. I've left some comment. PTAL

**Comment by DorianZheng** [comment]
reset LGTM

File: benches/coprocessor_executors.rs (hunk 1 above)
Comment (breeswish) [reply benches/coprocessor_executors.rs]: Sounds great! I will have a try.

**Comment by breeswish** [comment]
@DorianZheng SimpleBTreeMapStore has been removed now. Final data view is built over the RocksDB MVCC Storage. Please take a look again, thanks!

**Pull request merged**